### PR TITLE
Add example auth & functions

### PR DESCRIPTION
**Pull request by Cabskee**

### CODE REVIEW

File: database.rules.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (jordones) [inline database.rules.json]: This is sick, I didn't know you could put them in the project.

File: database.rules.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+  /* Visit https://firebase.google.com/docs/database/security to learn more about security rules. */
```
Comment (jack-bliss) [inline database.rules.json]: cant comment json files :(

Comment (jack-bliss) [reply database.rules.json]: or do we have a loader that allows it?

File: src/Games/selector.js [hunk 3]
```diff
@@ -31,6 +53,19 @@ export function GameSelector() {
     lastResult
   }, dispatch] = useReducer(gameSelectorReducer, gameSelectorInitialState);
 
+  const [user] = useUser();
+  const [homebase] = useState();
```
Comment (jack-bliss) [inline src/Games/selector.js]: missing setHomeBase?

File: functions/index.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
+const functions = require('firebase-functions');
+const firebase = require('firebase-admin');
+firebase.initializeApp();
+
+const userCollection = firebase.firestore().collection('users');
+const homebaseCollection = firebase.firestore().collection('homebase');
+const extensionsCollection = firebase.firestore().collection('extensions');
+
+exports.getHomebase = functions.https.onCall(async (data, {auth: {uid: userID} = {}}) => {
```
Comment (jack-bliss) [inline functions/index.js]: this is cool! like a repo for the frontend

File: src/api/homebase.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import {firebaseFunctions} from '../firebase';
```
Comment (jack-bliss) [inline src/api/homebase.js]: should this be `functions.js`?

File: database.rules.json (hunk 2 above)
Comment (Cabskee) [reply database.rules.json]: Because it's parsed by Firebase's JSON parser, comments are allowed

File: src/api/homebase.js (hunk 5 above)
Comment (Cabskee) [reply src/api/homebase.js]: Nope! Everything in `/functions` will not be bundled with the app, those actually get uploaded and become Lambda's - Then you use the frontend Firebase app (this `firebaseFunctions`) to call them, like lower in this file.

I'll eventually pull it out into its own repo (So we have separate FE/BE repo) but for now it was easier to show if it's in one

File: database.rules.json (hunk 1 above)
Comment (Cabskee) [reply database.rules.json]: Yeah, pretty awesome, this is how Firebase is starting to push everyone to do it. You can just edit these, run a verification test, then upload them straight from CLI

**Pull request merged**